### PR TITLE
Update delete/remove expense dialogs in edit expenses

### PR DESCRIPTION
**Pull request by yksolanki9**
Delete expense dialog - Before
Screenshot_20210831-143734

Delete expense dialog - After
https://user-images.githubusercontent.com/39493102/132223008-54d67867-6ce3-45ae-a4b2-d858e236fa5a.mp4

Remove expense dialog
https://user-images.githubusercontent.com/39493102/132223025-47696d80-bd0f-4749-851a-dd8e0dd17273.mp4

### CODE REVIEW

File: src/app/fyle/add-edit-expense/add-edit-expense.page.html [hunk 1]
```diff
@@ -43,7 +43,7 @@
       <ion-button class="add-edit-expense--header-btn-container__btn" (click)="openCommentsModal()">
         <ion-icon [src]="'../../../assets/svg/fy-chat-2.svg'" slot="icon-only"></ion-icon>
       </ion-button>
-      <ng-container *ngIf="etxn$|async as etxn">
+      <ng-container *ngIf="isExpenseDeletable && etxn$|async as etxn">
```
Comment (Dimple16) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.html:46]: `canDeleteExpense`

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts [hunk 2]
```diff
@@ -284,6 +284,8 @@ export class AddEditExpensePage implements OnInit {
 
   taxGroupsOptions$: Observable<{ label: string; value: any }[]>;
 
+  isExpenseDeletable = true;
```
Comment (Dimple16) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.ts:287]: same as above

**Review by Dimple16 — APPROVED** [review]

File: src/app/fyle/my-view-report/my-view-report.page.ts [hunk 3]
```diff
@@ -280,7 +281,7 @@ export class MyViewReportPage implements OnInit {
         {
           id: etxn.tx_id,
           navigate_back: true,
-          remove_from_report: true,
+          remove_from_report: erpt.rp_num_transactions > 1 ? true : false,
```
Comment (Julias0) [inline src/app/fyle/my-view-report/my-view-report.page.ts:284]: `remove_from_report: erpt.rp_num_transactions > 1`

File: src/app/fyle/add-edit-per-diem/add-edit-per-diem.page.ts [hunk 4]
```diff
@@ -212,7 +214,11 @@ export class AddEditPerDiemPage implements OnInit {
     private modalProperties: ModalPropertiesService
   ) {}
 
-  ngOnInit() {}
+  ngOnInit() {
+    if (this.activatedRoute.snapshot.params.remove_from_report) {
+      this.canDeleteExpense = this.activatedRoute.snapshot.params.remove_from_report === "true";
```
Comment (Julias0) [inline src/app/fyle/add-edit-per-diem/add-edit-per-diem.page.ts:219]: no `"` in ts files. Replace with 'true'

**Review by Dimple16 — APPROVED** [review]

**Pull request merged**